File: contracts/helloworld/helloworld.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package helloworld

import "github.com/iotaledger/wasp/packages/vm/wasmlib"

func funcHelloWorld(ctx *wasmlib.ScCallContext) {
	ctx.Log("Hello, world!")
}

func viewGetHelloWorld(ctx *wasmlib.ScViewContext) {
	ctx.Log("Get Hello world!")
	ctx.Results().GetString(VarHelloWorld).SetValue("Hello, world!")
}

File: contracts/helloworld/src/helloworld.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use wasmlib::*;

use crate::schema::*;

pub fn func_hello_world(ctx: &ScCallContext) {
    ctx.log("Hello, world!");
}

pub fn view_get_hello_world(ctx: &ScViewContext) {
    ctx.log("Get Hello world!");
    ctx.results().get_string(VAR_HELLO_WORLD).set_value("Hello, world!");
}